### PR TITLE
feat: add network api utils for crud operations

### PR DESCRIPTION
**Pull request by pprathameshaxioned**

### CODE REVIEW

File: src/utils/Api.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import axios from "axios";
```
Comment (sharshitprdxn) [inline src/utils/Api.js:1]: Rename this file to apiUtils

File: src/utils/Api.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return axiosInstance({
+    method: "POST",
+    url,
+    data,
+    headers: headers ? getHeaders() : {},
+  });
+};
+
+// delete is a reserved name
+export const del = (url) => {
```
Comment (sharshitprdxn) [inline src/utils/Api.js:35]: rename the to "delete"

File: src/utils/Api.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  };
+};
+
+export const post = (url, data, headers = true) => {
+  configAxios();
+  return axiosInstance({
+    method: "POST",
+    url,
+    data,
+    headers: headers ? getHeaders() : {},
```
Comment (sharshitprdxn) [inline src/utils/Api.js:30]: if the headers are false, we don't want a new header altogether overriding everything in the application, rather you can do is --- 
{...currentHeader, ...parameterHeaders}

File: src/utils/Api.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+let axiosInstance = axios.create({
+  baseURL: API_URL,
+});
+
+const configAxios = () => {
+  axiosInstance = axios.create({
+    baseURL: API_URL,
+  });
+};
+
```
Comment (sharshitprdxn) [inline src/utils/Api.js:15]: also include handling of the errors throughout... I have sent it on senior-dev channel

**Review by sharshitprdxn — CHANGES_REQUESTED** [review]
Trying setting in timeouts as well in the req/response objects.

File: src/utils/Api.js (hunk 1 above)
Comment (pprathameshaxioned) [reply src/utils/Api.js]: updated!

File: src/utils/Api.js (hunk 2 above)
Comment (pprathameshaxioned) [reply src/utils/Api.js]: updated!

File: src/utils/Api.js (hunk 3 above)
Comment (pprathameshaxioned) [reply src/utils/Api.js]: updated!

File: src/utils/Api.js (hunk 4 above)
Comment (pprathameshaxioned) [reply src/utils/Api.js]: changes are done.

**Review by sharshitprdxn — APPROVED** [review]

**Pull request merged**